packages/data-stores/src/auth/reducer.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { Reducer } from 'redux';
+import { combineReducers } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { LoginFlowState } from './types';
+import { Action } from './actions';
+
+export const loginFlowState: Reducer< LoginFlowState, Action > = (
+	state = 'ENTER_USERNAME_OR_EMAIL',
+	action
+) => {
+	switch ( action.type ) {
+		case 'RESET_LOGIN_FLOW':
+			return 'ENTER_USERNAME_OR_EMAIL';
+
+		case 'RECEIVE_AUTH_OPTIONS':
+			if ( ! action.response.passwordless ) {
+				return 'ENTER_PASSWORD';
+			}
+			return state;
+
+		case 'RECEIVE_WP_LOGIN':
+			return 'LOGGED_IN';
+
+		default:
+			return state;
+	}
+};
+
+const usernameOrEmail: Reducer< string, Action > = ( state = '', action ) => {
+	switch ( action.type ) {
+		case 'RESET_LOGIN_FLOW':
+			return '';
+
+		case 'RECEIVE_AUTH_OPTIONS':
+			return action.usernameOrEmail;
+
+		default:
+			return state;
+	}
+};
+
+const password: Reducer< string, Action > = ( state = '', action ) => {
```

Comment:
Do we need a password reducer? Looks like we're not (yet) storing anything here, and it seems like the `submitPassword` and `wpLogin` actions work nicely without it? (I always feel wary of storing passwords in state if we don't need to)

Reply:
Ya! I don't need this anymore, and I agree I don't want to naïvely JSON dump the store or something and leak it without realising.